firmware/signing.c

```diff
@@ -328,61 +334,45 @@ void phase2_request_next_input(void)
 
 void extract_input_bip32_path(const TxInputType *tinput)
 {
-	if (in_address_n_count == (size_t) -1) {
+	if (in_address_n_count == BIP32_NOCHANGEALLOWED) {
 		return;
 	}
 	size_t count = tinput->address_n_count;
-	if (count < 1) {
+	if (count < 2) {
```

Comment:
What about `if (count <= BIP32_NOCHANGEALLOWED)`?

Reply:
The logic here is that bip32 paths should follow the ..../0/x or .../1/x rule for wallets, so there should be at least two levels.  The change address should have the last two levels replaced with 1/x.

I want to keep the option to change BIP32_NOCHANGEALLOWED to something else (like -1), without having to change this line.

Reply:
What about adding `#define BIP32_MINLENGTH 2` or similar?

Reply:
Hmm, maybe something like `BIP32_WALLETDEPTH`, to indicate that this are the number of levels that change inside a single wallet. if we want to go this way, we should also have `#define BIP32_CHANGECHAIN 1`.  It is still not fully configurable, e.g. in `check_change_bip32_path` we need to check each of these two levels for validity.